### PR TITLE
Updates in Docs

### PR DESCRIPTION
**Pull request by honzajavorek**
Mission Objective: Make Mr. Swagger feel like first-class citizen in the Dredd Documentation.

------

**Please ignore AppVeyor, it's not properly set up and actively used yet.**

### CODE REVIEW

File: docs/example.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE HTML>
```
Comment (honzajavorek) [inline docs/example.html]: This is just workaround for Read The Docs to redirect `/example` to `/quickstart`. I merged those two pages.

File: docs/example.md [hunk 2]
```diff
@@ -1,64 +0,0 @@
-# Dredd Example with API Backend Application
```
Comment (honzajavorek) [inline docs/example.md]: I merged this page with `/quickstart`.

File: docs/example/index.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE HTML>
```
Comment (honzajavorek) [inline docs/example/index.html]: This is just workaround for Read The Docs to redirect `/example` to `/quickstart`. I merged those two pages.

File: docs/execution-lifecycle.html [hunk 4]
```diff
@@ -11,7 +11,7 @@
         <meta charset="UTF-8">
         <meta http-equiv="refresh" content="0;url=how-it-works/#execution-life-cycle">
         <script type="text/javascript">
-            window.location.href = 'how-it-works/#execution-life-cycle' + window.location.hash
+            window.location.href = 'how-it-works/#execution-life-cycle'
```
Comment (honzajavorek) [inline docs/execution-lifecycle.html]: This is bug (the URL already has it's own `#hash`).

File: docs/execution-lifecycle/index.html [hunk 5]
```diff
@@ -11,14 +11,14 @@
         <meta charset="UTF-8">
         <meta http-equiv="refresh" content="0;url=../how-it-works/#execution-life-cycle">
         <script type="text/javascript">
-            window.location.href = '../how-it-works/#execution-life-cycle' + window.location.hash
+            window.location.href = '../how-it-works/#execution-life-cycle'
```
Comment (honzajavorek) [inline docs/execution-lifecycle/index.html]: This is bug (the URL already has it's own `#hash`).

File: docs/index.md [hunk 6]
```diff
@@ -42,27 +42,23 @@ Dredd supports writing [hooks](https://dredd.readthedocs.io/en/latest/hooks/)
...
+- [Quickstart](quickstart.md)
+- [How It Works](how-it-works.md)
+- [How-To Guides](how-to-guides.md)
 - Usage
     - [Command-line Interface](usage-cli.md)
-    - [From JavaScript](usage-js.md)
+    - [As a JavaScript Library](usage-js.md)
 - Hooks
     - [About Hooks](hooks.md)
-    - Supported Languages
```
Comment (honzajavorek) [inline docs/index.md]: This was way too nested in the left column, making it more flat.

File: mkdocs.yml [hunk 7]
```diff
@@ -4,27 +4,23 @@ site_description: 'HTTP API Testing Framework'
 site_author: 'Apiary'
 
 pages:
```
Comment (honzajavorek) [inline mkdocs.yml]: These items were outdated for longer time and I noticed it only now.

File: docs/quickstart.md [hunk 8]
```diff
@@ -1,22 +1,102 @@
...
+At this moment, the implementation is ready to be tested. Let's run the server as a background process and let's test it:
+
+```sh
+$ node app.js &
+```
+
+Finally, let Dredd validate whether your freshly implemented API complies with the description you have:
+
+```sh
+$ dredd api-description.apib http://localhost:3000  # API Blueprint
```
Comment (miiila) [inline docs/quickstart.md]: I am not so familiar with Dredd params, so just ensure me, it won't start interactive config :-)

**Comment by miiila** [comment]
OK from my point view, but I have to admit I haven't followed the tutorial and just read the content.

File: docs/quickstart.md (hunk 8 above)
Comment (honzajavorek) [inline docs/quickstart.md]: No, only `dredd init` starts interactive config.

**Comment by honzajavorek** [comment]
Thanks!

**Pull request merged**